server-common/src/test/java/org/apache/kafka/server/util/MockTime.java

```diff
@@ -27,15 +23,21 @@
  * 1. This has an associated scheduler instance for managing background tasks in a deterministic way.
  * 2. This doesn't support the `auto-tick` functionality as it interacts badly with the current implementation of `MockScheduler`.
  */
-class MockTime(currentTimeMs: Long, currentHiResTimeNs: Long) extends JMockTime(0, currentTimeMs, currentHiResTimeNs) {
-
-  def this() = this(System.currentTimeMillis(), System.nanoTime())
+public class MockTime extends org.apache.kafka.common.utils.MockTime {
+    public final MockScheduler scheduler;
```

Comment:
Can we make this private and add an accessor? Or would that break a bunch of usages? I think Scala usages might be ok with the a no-parens method call

Reply:
It would break a bunch of java usages. This is why I kept it like this.